source/computer/commands.rst

```diff
@@ -63,54 +62,557 @@ Linux/macOS 下有成百上千个命令，每个命令都有众多选项。这
+命令，但是它不占资源，不会提高多少系统负荷。当有许多小文件需要传输时，使用 `rsync`_ 命名会导致
+硬盘 I/O（输入/输出）非常高，而 ``scp`` 基本不影响系统正常使用。
+
+以下命令假定远程电脑的 IP 地址是 192.168.1.100，用户名是 seismo-learn。
+
+复制远程文件 :file:`/home/seismo-learn/fk3.3.tar.gz` 到本地目录 :file:`~/Downloads` 下::
+
+    $ scp seismo-learn@192.168.1.100:/home/seismo-learn/fk3.3.tar.gz ~/Downloads/
+
+复制远程目录 :file:`/home/seismo-learn/folder1` 到本地目录 :file:`~/Downloads` 下::
+
+    $ scp -r seismo-learn@192.168.1.100:/home/seismo-learn/folder1 ~/Downloads/
+
+上传本地文件 :file:`~/Downloads/fk3.3.tar.gz` 到远程目录 :file:`home/seismo-learn/folder2` 下::
+
+    $ scp ~/Downloads/fk3.3.tar.gz seismo-learn@192.168.1.100:/home/seismo-learn/folder2/
+
+上传本地目录 :file:`~/Downloads/folder1` 到远程目录 :file:`home/seismo-learn/folder2` 下::
+
+    $ scp ~/Downloads/folder1 seismo-learn@192.168.1.100:/home/seismo-learn/folder2/
+
+sed
+---
+
+``sed`` 命令的名字来源与 **s**\ tream **ed**\ itor（流编辑器）。该命令可以用于对输入流
+（文件或管道）执行基本的文本转换。它会把当前处理的行存储在临时缓冲区中再进行处理，处理完成后
+再把缓冲区的内容送往屏幕。接着处理下一行，直到文件末尾。因此默认情况下，文件内容并没有改变。
+
+将 :file:`file` 中每一行的第一个 book 替换成 books::
+
+    $ sed 's/book/books/' file
+
+将 :file:`file` 中每一行的所有的 book 都替换成 books::
+
+    $ sed 's/book/books/g' file
+
+以上命令只是将转换后的文本内容打印出来，并为改变文件本身。可以使用 ``-i`` 选项直接改变文件::
+
+    $ sed -i 's/book/books/g' file
+
+以上命令使用斜杠 :kbd:`/` 当定界符，也可以使用任意定界符::
+
+    $ sed 's#book#books#' file
+    $ sed 's#book#books#g' file
+    $ sed -i 's#book#books#g' file
+
 sort
 ----
 
+``sort`` 命令可以将文件内容进行排序，并打印排序结果。该命令将文件的每一行作为一个单位，相互比较。
+默认的比较原则是从首字符向后，依次按 ASCII 码值进行比较，最后将他们按排序结果输出。
+
+我们使用示例文件 :file:`seismo-learn-sort.txt` 展示该命令的主要用法::
+
+    $ cat seismo-learn-sort.txt
+    6:software:seisman:-1.3
+    1:seismology101:zhaozhiyuan1989:291
+    7:software:core-man:101.1
+    1:seismology101:zhaozhiyuan1989:291
+    2:seismology101:seisman:80
+    3:seismology101:wangliang1989:101.2
+    8:software:zhaozhiyuan1989:291
+    5:seismology:core-man:-81.2
+    4:seismology:seisman:91
+    1:seismology101:zhaozhiyuan1989:291
+
+按 ASCII 码值进行升序排序::
+
+    $ sort seismo-learn-sort.txt
+
+按 ASCII 码值进行降序排序::
+
+    $ sort -r seismo-learn-sort.txt
+
+按 ASCII 码值进行升序排序，并忽略相同行（即重复行只统计一次）::
+
+    $ sort -u seismo-learn-sort.txt
+
+按 ASCII 码值进行降序排序，并忽略相同行::
+
+    $ sort -u -r seismo-learn-sort.txt
+    # 不同选项也可以写在一块
+    # sort -ur seismo-learn-sort.txt
+
+按\ **数值大小**\ 进行升序排序::
+
+    $ sort -n seismo-learn-sort.txt
+
+使用冒号 :kbd:`：` 作为字段分隔符，并从第二个字段的首个字符到行末，升序排序 :file:`seismo-learn-sort.txt` ::
+
+    $ sort -t ":" -k 2 seismo-learn-sort.txt
+
+同上，但使用降序排序（以下两种方式均可）::
+
+    $ sort -t ":" -k 2r seismo-learn-sort.txt
+
+只按照第二个字段进行降序排序::
+
+    $ sort -t ":" -k 2,2r seismo-learn-sort.txt
+
+还可以多次使用 ``-k`` 选项，这样会在前一个选项指定的字符范围的排序结果相同时，
+接着比较下一个选项指定的字符范围。
+
+按照第二个字段进行降序排序，若相同则按照第三个字段升序排序::
+
+    $ sort -t ":" -k 2,2r -k 3,3 seismo-learn-sort.txt
+
+按照第二个字段进行降序排序，若相同则按照第四个字段的数值大小降序排序::
+
+    $ sort -t ":" -k 2,2r -k 4,4nr seismo-learn-sort.txt
+
+同上，但并忽略相同行::
+
+    $ sort -t ":" -k 2,2r -k 4,4nr -u seismo-learn-sort.txt
```

Comment:
This part is complicated and can be removed.

Reply:
done in 777a154.